app/Http/Controllers/Admin/DigitalPublicationArticleController.php

```diff
@@ -62,12 +72,43 @@ protected function indexData($request)
         ];
     }
 
+    protected function transformIndexItems($items)
+    {
+        // If we're in the browser, don't transform the items
+        if (property_exists($items, 'path')) {
+            return $items;
+        }
+        return $items->toTree();
+    }
+
+    protected function indexItemData($item)
+    {
+        return $item->children ? [
+            'children' => $this->getIndexTableData($item->children),
+        ] : [];
+    }
+
+    private function getParents($exceptPage = null)
+    {
+        return $this->repository->whereNotIn('id', is_null($exceptPage) ? [] : [$exceptPage])->withDepth()->defaultOrder()->orderBy('position')->get()->filter(function ($page) {
+            return $page->depth < 3;
+        })->values();
+    }
+
     protected function formData($request)
     {
         $item = $this->repository->getById(request('article') ?? request('id'));
         $digPub = app(DigitalPublicationRepository::class)->getById(request('digitalPublication'));
         $baseUrl = '//' . config('app.url') . '/' . $this->permalinkBase . $digPub->id . '/' . $digPub->getSlug() . '/' . $item->id . '/';
 
+        $ancestors = $item->ancestors()->defaultOrder()->get();
+
+        $baseUrl = '//' . config('app.url') . '/';
+
+        foreach ($ancestors as $ancestor) {
+            $baseUrl = $baseUrl . $ancestor->slug . '/';
```

Comment:
You can use the [`url` helper](https://laravel.com/docs/10.x/helpers#method-url) to construct these urls: `$baseUrl = url($ancestor->slug)`